### DISCUSSION TITLE
Source generators - prevent running source generator while typing in Visual Studio

### DISCUSSION THREAD

**Discussion by zbesevic** [dotnet/roslyn#52604] [General]
Hi there,

Is there a way to stop running the source generator while editing files in visual studio?

I would need to run the generator only when explicitly compiling.

**Comment by jmarolf** [comment]
Therre is no api for this today. It is expected that you use an msbuild task if you want fine grained control of when you run.

**Reply by zbesevic** [reply]
Can you give me an example of this I cannot figure it out?

What I did is to write a file in the prebuild event which is triggered when compile comes not from typing and i could fix it. But i cannot see the generated files anymore in that case. I have the performance back but i cannot see the files which is not good enough for me.
